dali/base/dasess.cpp

```diff
@@ -947,9 +947,11 @@ class CClientSessionManager: public CSessionManagerBase, implements IConnectionM
             {
                 CDateTime now;
                 StringBuffer b64sig;
-                createDaliSignature(obj, udesc, now, b64sig);
-                mb.append(b64sig.str());
-                now.serialize(mb);
+                if (createDaliSignature(obj, udesc, now, b64sig))
+                {
+                    mb.append(b64sig.str());
+                    now.serialize(mb);
+                }
```

Comment:
Can the code handles the missing signature when the mb is read?

Reply:
Yes, in processMessage, the code reads only if there is more

                StringBuffer reqSignature;
                CDateTime reqUTCTimestamp;
                if (mb.remaining() > 0)
                {
                    mb.read(reqSignature);
                    reqUTCTimestamp.deserialize(mb);
                }